enterprise/views.py

```diff
@@ -0,0 +1,184 @@
+    render_to_response = None
+
+try:
+    from edxmako.paths import add_lookup
+except ImportError:
+    add_lookup = None
+
+try:
+    from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+except ImportError:
+    configuration_helpers = None
+
+try:
+    from third_party_auth.pipeline import get_complete_url, get_real_social_auth_object
+except ImportError:
+    get_complete_url = None
+    get_real_social_auth_object = None
+
+from enterprise.models import EnterpriseCustomer, EnterpriseCustomerUser, UserDataSharingConsentAudit
+from enterprise.tpa_pipeline import active_provider_enforces_data_sharing, get_enterprise_customer_for_request
+from enterprise.utils import NotConnectedToEdX
+
+
+class GrantDataSharingPermissions(View):
+    """
+    Provide a form and form handler for data sharing consent.
+
+    View handles the case in which we get to the "verify consent" step, but consent
+    hasn't yet been provided - this view contains a GET view that provides a form for
+    consent to be provided, and a POST view that consumes said form.
+    """
+
+    @staticmethod
+    def patch_mako_lookup():
+        """
+        Update the EdX Mako paths to point to our consent template.
+        """
+        if add_lookup is None:
+            raise NotConnectedToEdX(
+                _('Must be installed in an OpenEdX environment to render this response.')
+            )
+        full_location = os.path.realpath(__file__)
+        directory = os.path.dirname(full_location)
+        template_location = os.path.join(directory, 'templates', 'enterprise')
+        add_lookup('main', template_location)
+
+    @staticmethod
+    def quarantine(request):
+        """
+        Set a session variable to quarantine the user to ``enterprise.views``.
+        """
+        request.session['sso_quarantined_modules'] = (
+            'enterprise.views',
+        )
+
+    @staticmethod
+    def lift_quarantine(request):
+        """
+        Remove the quarantine session variable.
+        """
+        request.session.pop('sso_quarantined_modules', None)
+
+    @staticmethod
+    def get_warning(provider, platform, required):
+        """
+        Get the appropriate warning for the form.
+        """
+        if required:
+            return _(
+                "Are you sure?\nIf you do not agree to share your data, you will have to use "
+                "another account to access {platform}."
+            ).format(platform=platform)
+        else:
+            return _(
+                "Are you sure?\nIf you do not agree to share your data, you will not get any "
+                "discounts from {provider}."
+            ).format(provider=provider)
+
+    @staticmethod
+    def get_note(provider, required):
+        """
+        Get the appropriate note for the form.
+        """
+        if required:
+            return _(
+                "*{provider} requires data sharing consent; if consent is not provided, you will"
```

Comment:
Why is the asterisk inside this string? It will be confusing to translators, and it is confusing to me. If it's supposed to be a bullet or something, add it later at the template level.